CollapseLauncher/Classes/InstallManagement/BaseClass/InstallManagerBase.cs

```diff
@@ -481,17 +483,168 @@ public async Task MoveGameLocation()
+                        }
+
+                        // Cleanup Game_Data folder while keeping whatever specified in foldersToKeepInData
+                        foreach (string folderGameData in Directory.GetDirectories(Path.Combine(GameFolder, gameDataFolderName)))
+                        {
+                            try
+                            {
+                                if (foldersToKeepInData != null && !foldersToKeepInData.Contains(folderGameData)) // Skip this entire process if foldersToKeepInData is null
+                                {
+                                    if (File.GetAttributes(folderGameData).HasFlag(FileAttributes.Directory))
+                                    {
+                                        Directory.Delete(folderGameData, true);
+                                        LogWriteLine($"Deleted folder: {folderGameData}", LogType.Warning, true);
+                                    }
+                                    else
+                                    {
+                                        File.Delete(folderGameData);
+                                        LogWriteLine($"Deleted file: {folderGameData}", LogType.Default, true);
+                                    }
+                                    continue;
+                                }
+                                // Do check if the folder empty after cleaning up 
+                                if (Directory.Exists(folderGameData) && !Directory.EnumerateFileSystemEntries(folderGameData).Any())
+                                {
+                                    Directory.Delete(folderGameData);
+                                    LogWriteLine($"Deleted empty game folder: {folderGameData}", LogType.Default, true);
+                                }
+                            }
+                            catch (Exception ex)
+                            {
+                                LogWriteLine($"An error occurred while deleting object {folderGameData}\r\n{ex}", LogType.Error, true);
+                            }
+                        }
+
+                        // Cleanup any folders in foldersToDelete
+                        foreach (string folderName in Directory.GetDirectories(GameFolder))
+                        {
+                            if (foldersToDelete != null && foldersToDelete.Contains(Path.GetFileName(folderName)))
+                            {
+                                try
+                                {
+                                    Directory.Delete(folderName, true);
+                                    LogWriteLine($"Deleted {folderName}", LogType.Default, true);
+                                }
+                                catch (Exception ex)
+                                {
+                                    LogWriteLine($"An error occurred while deleting folder {folderName}\r\n{ex}", LogType.Error, true);
```

Comment:
Maybe consider reverting delete operation & attempt to restore files/directories if possible?

Reply:
that would require moving the folder first to another location and revert everything when it found an error, though I don't see this being very useful as the intention was to delete the entire folder anyway, this error shouldn't happen unless the folder in foldersToDelete doesn't exist, but since it uses `continue;` it will keep deleting the other folder instead of failing

Reply:
Digging a little deeper, it is indeed not feasible to restore files through C# natively, therefore I think we can leave it as is: https://stackoverflow.com/questions/8819188/c-sharp-classes-to-undelete-files